### PR TITLE
Add isActivatedBy to ShortcutActivator

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

When I was working on the menu bar, I needed a simple way to tell if an event would activate a shortcut activator in the same way that `Shortcuts` does it.

## Tests
 - Added tests for all the affected classed.

### CODE REVIEW

**Comment by dkwingsmt** [comment]
I'm not in favor of this change. I don't think this adds anything that couldn't be done previously. The subclass overrides are actually unnecessary since they're all equivalent to the base implementation
```dart
  bool isActivatedBy(RawKeyEvent event, [RawKeyboard? state]) {
    return (triggers?.contains(event.logicalKey) ?? true)
        && accepts(event, state ?? RawKeyboard.instance);
  }
```
It saves a few letters to type, but does not justify the cost of adding a new method to a class that is meant to be extended.

**Comment by gspencergoog** [comment]
Tong and I spoke offline and decided to move this to be a static function on `ShortcutActivator` instead.

**Comment by gspencergoog** [comment]
@dkwingsmt OK, updated to be a static function. PTAL.

File: packages/flutter/lib/src/widgets/shortcuts.dart [hunk 1]
```diff
@@ -211,6 +211,16 @@ abstract class ShortcutActivator {
   ///    modifier key is pressed when the side variation is not important.
   bool accepts(RawKeyEvent event, RawKeyboard state);
 
+  /// Returns true if the event and keyboard state would cause this
+  /// [ShortcutActivator] to be activated.
+  ///
+  /// If the keyboard `state` isn't supplied, then it defaults to using
+  /// [RawKeyboard.instance].
+  static bool isActivatedBy(ShortcutActivator activator, RawKeyEvent event, [RawKeyboard? state]) {
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/widgets/shortcuts.dart:219]: Why do we have an optional `state`? I don't see anywhere omitting it. Unless we're using the omitted version, let's enforce it and write "typically you should use" in the doc.

**Review by dkwingsmt — APPROVED** [review]
A small request, otherwise LGTM.

File: packages/flutter/lib/src/widgets/shortcuts.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/shortcuts.dart:219]: It's not used in this PR, but I'm planning on using the omitted version in the menu code. If it weren't for needing to supply it for tests, I would just eliminate the optional parameter.

**Pull request merged**